Content.Client/UserInterface/Controls/RadialMenu.cs

```diff
@@ -205,51 +278,308 @@ private void OnClicked(ButtonEventArgs args)
+                ? HoverBorderColor
+                : BorderColor;
+            DrawAnnulusSector(handle, containerCenter, _innerRadius, _outerRadius, angleFrom, angleTo, borderColor, false);
+        }
+
+        if (!_isWholeCircle && DrawSeparators)
+        {
+            DrawSeparatorLines(handle, containerCenter, _innerRadius, _outerRadius, angleFrom, angleTo, SeparatorColor);
+        }
     }
 
-    private RadialMenu? FindParentMultiLayerContainer(Control control)
+    /// <inheritdoc />
+    protected override bool HasPoint(Vector2 point)
     {
-        foreach (var ancestor in control.GetSelfAndLogicalAncestors())
+        if (_parentCenter == null)
         {
-            if (ancestor is RadialMenu)
-                return ancestor as RadialMenu;
+            return base.HasPoint(point);
         }
 
-        return null;
+        var outerRadiusSquared = _outerRadius * _outerRadius;
+        var innerRadiusSquared = _innerRadius * _innerRadius;
+
+        var distSquared = (point + Position - _parentCenter.Value).LengthSquared();
+        var isInRadius = distSquared < outerRadiusSquared && distSquared > innerRadiusSquared;
+        if (!isInRadius)
+        {
+            return false;
+        }
+
+        // difference from the center of the parent to the `point`
+        var pointFromParent = point + Position - _parentCenter.Value;
+
+        // Flip Y to get from ui coordinates to natural coordinates
+        var angle = MathF.Atan2(-pointFromParent.Y, pointFromParent.X) - _angleOffset;
+        if (angle < 0)
+        {
+            // atan2 range is -pi->pi, while angle sectors are
+            // 0->2pi, so remap the result into that range
+            angle = MathF.PI * 2 + angle;
+        }
+
+        var isInAngle = angle >= _angleSectorFrom && angle < _angleSectorTo;
+        return isInAngle;
+    }
+
+    /// <summary>
+    /// Draw segment between two concentrated circles from and to certain angles.
+    /// </summary>
+    /// <param name="drawingHandleScreen">Drawing handle, to which rendering should be delegated.</param>
+    /// <param name="center">Point where circle center should be.</param>
+    /// <param name="radiusInner">Radius of internal circle.</param>
+    /// <param name="radiusOuter">Radius of external circle.</param>
+    /// <param name="angleSectorFrom">Angle in radian, from which sector should start.</param>
+    /// <param name="angleSectorTo">Angle in radian, from which sector should start.</param>
+    /// <param name="color">Color for drawing.</param>
+    /// <param name="filled">Should figure be filled, or have only border.</param>
+    private void DrawAnnulusSector(
+        DrawingHandleScreen drawingHandleScreen,
+        Vector2 center,
+        float radiusInner,
+        float radiusOuter,
+        float angleSectorFrom,
+        float angleSectorTo,
+        Color color,
+        bool filled = true
+    )
+    {
+        const float minimalSegmentSize = MathF.Tau / 32;
```

Comment:
```suggestion
        const float minimalSegmentSize = MathF.Tau / 128f;
```

You can afford some extra triangles

Reply:
Well if you say so. It was looking KINDA ugly. done